Cocktails9/Cocktails9/Preview Content/GridItemView.swift

```diff
@@ -0,0 +1,47 @@
+import SwiftUI
+
+struct GridItemView: View {
+    @State var drink: Drink
+    @State var user: User
+    
+    var body: some View {
+        VStack(alignment: .leading) {
+            AsyncImage(url: URL(string: drink.imageUrl)) { image in
+                image.resizable().scaledToFit()
+            } placeholder: {
+                ProgressView()
+            }
+            .frame(width: 155, height: 155)
+            .clipShape(RoundedRectangle(cornerRadius: 10))
+            
+            // HStack to hold both the drink name and favorite button
+            HStack {
+                Text(drink.name)
+                    .font(.headline)
+                    .multilineTextAlignment(.center)
+                    .lineLimit(2)
+                    .padding(.top, 5)
+                    .frame(maxWidth: 150) // Adjust max width if necessary
+                    .foregroundColor(.black)
+
+                Spacer()  // Push the favorite button to the right side
+                
+                Button(action: {
+                    drink.toggleFavourite()
+                    
+                    if drink.isFavourite {
+                        user = UserManager.addFavoriteDrink(to: user, drink: drink)
+                    } else {
+                        user = UserManager.removeFavoriteDrink(from: user, drink: drink)
+                    }
+                }) {
+                    Image(systemName: drink.isFavourite ? "heart.fill" : "heart")
+                        .foregroundColor(drink.isFavourite ? .red : .gray)
+                        //.padding(5)  
+                }
```

Comment:
I'm guessing this implementation has a bug in drink is not saved to favourites on first tap 🙂  To fix this, you'll need to rewrite the logic.
Second thing is to move a `isFavourite` getter from `Drink` struct to this struct. It can be local or global private variable. Doesn't matter. Either way, you need a different way to check if the drink is favourite or not.